package-checks.cake

```diff
@@ -0,0 +1,237 @@
+            HasDirectory("lib/netstandard1.6").WithFile("nunit.engine.api.dll"),
+            HasDirectory("lib/netstandard2.0").WithFile("nunit.engine.api.dll")) &
+        CheckNuGetPackage(
+            "NUnit.Runners",
+            HasFile("LICENSE.txt")) &
+        CheckChocolateyPackage(
+            "nunit-console-runner",
+            HasDirectory("tools").WithFiles("LICENSE.txt", "NOTICES.txt", "CHANGES.txt", "VERIFICATION.txt").AndFiles(CONSOLE_FILES).AndFiles(ENGINE_FILES),
+            HasDirectory("tools/agents/net20").WithFiles(AGENT_FILES),
+            HasDirectory("tools/agents/net40").WithFiles(AGENT_FILES)) &
+        CheckChocolateyPackage(
+            "nunit-console-with-extensions",
+            HasDirectory("tools").WithFiles("LICENSE.txt", "NOTICES.txt", "VERIFICATION.txt")) &
+        CheckZipPackage(
+            "NUnit.Console",
+            HasFiles("LICENSE.txt", "license.rtf", "NOTICES.txt", "CHANGES.txt"),
+            HasDirectory("bin/net20").WithFiles("nunit3-console.exe", "nunit3-console.exe.config").AndFiles(ENGINE_FILES),
+            HasDirectory("bin/net35").WithFiles("nunit3-console.exe", "nunit3-console.exe.config").AndFiles(ENGINE_FILES),
+            HasDirectory("bin/netstandard1.6").WithFiles(ENGINE_FILES),
+            HasDirectory("bin/netstandard2.0").WithFiles(ENGINE_FILES),
+            HasDirectory("bin/netcoreapp1.1").WithFiles(ENGINE_FILES),
+            HasDirectory("bin/netcoreapp1.1").WithFiles(ENGINE_FILES),
+            HasDirectory("bin/agents/net20").WithFiles(AGENT_FILES),
+            HasDirectory("bin/agents/net40").WithFiles(AGENT_FILES)) &
+        CheckMsiPackage("NUnit.Console", HasDirectory("NUnit.org/nunit-console")); // Placeholder for later implementation
+
+    if (!isOK)
+        throw new Exception("One or more package checks failed. See listing.");
+}
+
+private bool CheckNuGetPackage(string packageId, params ICheck[] checks)
+{
+    return CheckPackage($"{PACKAGE_DIR}{packageId}.{productVersion}.nupkg", checks);
+}
+
+private bool CheckChocolateyPackage(string packageId, params ICheck[] checks)
+{
+    return CheckPackage($"{PACKAGE_DIR}{packageId}.{productVersion}.nupkg", checks);
+}
+
+// NOTE: Zip and msi packages currently use "version" rather than "productVersion"
+
+private bool CheckZipPackage(string packageName, params ICheck[] checks)
+{
+    return CheckPackage($"{PACKAGE_DIR}{packageName}-{version}.zip", checks);
+}
+
+private bool CheckMsiPackage(string packageName, params ICheck[] checks)
+{
+    return CheckPackage($"{PACKAGE_DIR}{packageName}-{version}.msi", checks);
+}
+
+private bool CheckPackage(string path, params ICheck[] checks)
+{
+    Console.WriteLine("\nPackage Name: " + System.IO.Path.GetFileName(path));
+
+    if (!FileExists(path))
+    {
+        WriteError("Package was not found!");
+        return false;
+    }
+
+    bool allOK = true;
+
+    // TODO: Figure out how to check content of msi
```

Comment:
We could shell out to msiexec:

```cs
var exitCode = StartProcess(
    "msiexec",
    new ProcessArgumentBuilder()
        .Append("/a")
        .AppendQuoted(path)
        .AppendQuoted("TARGETDIR=" + someTemporaryDirectory)
        .Append("/quiet")
        .Render());
```

Reply:
https://docs.microsoft.com/en-us/windows/win32/msi/command-line-options

Reply:
@jnm2 Yup, we could install it. I was trying to figure out how to do it without installing, however, as I do for the nuget and chocolatey packages. I did it that way because you need to run as administrator to install a chocolatey package.

I also hesitate to get too deep into the msi, with which I'm no longer very familiar.

Reply:
I didn't realize this would install the msi. Are you sure it does?

Reply:
Well, /a is the administrative install option, which basically creates a share from which others install. It may be worth trying so as to get a directory where a check could be performed.

Reply:
Yeah, that works! It gives me a temp directory with a copy of the msi __and__ an NUnit.org directory containing the entire installation. So, I guess I'll add it to the checks.

Reply:
@jnm2 OK... it works programmatically... __finally__!

I decided to just roll my own with strings rather than using `ProcessArgumentBuilder`... 'cause... that's me! :smiley: 

In terms of efficiency, probably not the best idea but actually a good one for learning. I am amazed to discover that `msiexec` does not tolerate forward slashes in paths. And the error message it gave me about an invalid character replaced the '/' with a '.' because (apparently) it things a slash is non-representable!

Yet another reason to hate the whole msi thing!